src/prometheus/prometheus_launcher.star

```diff
@@ -64,7 +65,74 @@ def get_config(config_files_artifact_name):
 	)
 
 
-def new_config_template_data(cl_nodes_metrics_info):
+def new_config_template_data(
+	el_client_contexts,
+	cl_client_contexts,
+	additional_metrics_jobs,
+):
+	metrics_jobs = []
+	# Adding execution clients metrics jobs
+	for context in el_client_contexts:
+		if len(context.el_metrics_info) >= 1 and context.el_metrics_info[0] != None:
+			execution_metrics_info = context.el_metrics_info[0]
+			metrics_jobs.append(new_metrics_job(
+				job_name = execution_metrics_info["name"],
+				endpoint = execution_metrics_info["url"],
+				metrics_path = execution_metrics_info["path"],
+				labels = {
+					"service": context.service_name,
+					"client_type": "execution",
+					"client_name": context.client_name,
+				},
+			))
+	# Adding consensus clients metrics jobs
+	for context in cl_client_contexts:
+		if len(context.cl_nodes_metrics_info) >= 1 and context.cl_nodes_metrics_info[0] != None:
+			# Adding beacon node metrics
+			beacon_metrics_info = context.cl_nodes_metrics_info[0]
+			metrics_jobs.append(new_metrics_job(
+				job_name = beacon_metrics_info["name"],
+				endpoint = beacon_metrics_info["url"],
+				metrics_path = beacon_metrics_info["path"],
+				labels = {
+					"service": context.beacon_service_name,
+					"client_type": "beacon",
+					"client_name": context.client_name,
+				},
+			))
+		if len(context.cl_nodes_metrics_info) > 1 and context.cl_nodes_metrics_info[1] != None:
+			# Adding validator node metrics
+			validator_metrics_info = context.cl_nodes_metrics_info[1]
+			metrics_jobs.append(new_metrics_job(
+				job_name = validator_metrics_info["name"],
+				endpoint = validator_metrics_info["url"],
+				metrics_path = validator_metrics_info["path"],
+				labels = {
+					"service": context.validator_service_name,
+					"client_type": "validator",
+					"client_name": context.client_name,
+				},
+			))
+	# Adding additional metrics jobs
+	for job in additional_metrics_jobs:
+		if job == None:
+			continue
+		metrics_jobs.append(job)
+	return {
+		"MetricsJobs": metrics_jobs,
+	}
+
+def new_metrics_job(
+	job_name,
+	endpoint,
+	metrics_path,
+	labels,
+	scrape_interval = "",
+):
 	return {
-		"CLNodesMetricsInfo": cl_nodes_metrics_info
+		"Name": job_name,
+		"Endpoint": endpoint,
+		"MetricsPath": metrics_path,
+		"Labels": labels,
+		"ScrapeInterval": scrape_interval,
```

Comment:
does setting ScrapeInterval to "" work? Or should we set it to a reasonable default?

Reply:
Say 15s like below

Reply:
Its ignored inside the template if empty. In those cases uses default 15s. I guess adding 15s on default is fine too.